### DISCUSSION TITLE
classes/class_vector3

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#219] [User-contributed notes]
# classes/class_vector3

A 3D vector using floating-point coordinates. Description: A 3-element structure that can be used to represent 3D coordinates or any other triplet of numeric values. It uses floating-point coordina...

https://docs.godotengine.org/en/stable/classes/class_vector3.html

**Comment by donn-xx** [comment]
For **Project** and **Slide**, see this image on Wikipedia: https://upload.wikimedia.org/wikipedia/commons/f/f7/Projection_and_rejection_2.png
Given vectors a and b:
a1 = a.project(b)
a2 = a.slide(b)
